src/CommonITILObject.php

```diff
@@ -7951,10 +7951,20 @@ public function getForbiddenSingleMassiveActions()
     /**
      * Returns criteria that can be used to get documents related to current instance.
      *
+     * @param bool      $bypass_rights  Whether to bypass rights checks (default: false)
+     * @param User|null $user           User for rights checking (default: null = current session rights)
+     *
      * @return array
      */
-    public function getAssociatedDocumentsCriteria($bypass_rights = false): array
+    public function getAssociatedDocumentsCriteria($bypass_rights = false, ?User $user = null): array
     {
+        $user_id = $user ? $user->getID() : Session::getLoginUserID();
+
+        if ($user === null) {
+            $user = new User();
+            $user->getFromDB(Session::getLoginUserID());
+        }
```

Comment:
Using the `$user` object instead of the session will add, at least, 5 SQL queries more to display the ticket timeline. It should keep the checks on session when the `$user` param is null.

```php
if (
    !$bypass_rights
    && (
        ($user === null && !Session::haveRight(ITILFollowup::$rightname, ITILFollowup::SEEPRIVATE))
        || ($user !== null && !$user->hasRight(ITILFollowup::$rightname, ITILFollowup::SEEPRIVATE))
    )
) {
```